pkg/processor/runtime/rpc/connection/abstract.go

```diff
@@ -67,6 +72,113 @@ func (bc *AbstractConnectionManager) SetStatus(newStatus status.Status) {
 	//bc.abstractRuntime.SetStatus(newStatus)
 }
 
+// PrepareControlMessageSocket prepares control message socket for processing
+// If SupportControlCommunication is enabled, a control communication socket is created,
+// wrapped in a ControlMessageSocket, and integrated with the ControlMessageBroker for runtime operations.
+func (bc *AbstractConnectionManager) prepareControlMessageSocket() error {
+	if bc.Configuration.SupportControlCommunication {
+		controlConnection, err := bc.createSocketConnection()
+		if err != nil {
+			return errors.Wrap(err, "Failed to create control socket connection")
+		}
+		bc.controlMessageSocket = NewControlMessageSocket(
+			bc.Logger,
+			controlConnection,
+			bc.RuntimeConfiguration.ControlMessageBroker)
+	}
+	return nil
+}
+
+func (bc *AbstractConnectionManager) startControlMessageSocket() error {
+	if bc.Configuration.SupportControlCommunication {
```

Comment:
```suggestion
	if bc.Configuration.SupportControlCommunication && bc.controlMessageSocket != nil {
```

Reply:
@TomerShor at this point it will always be nil